meta-refkit/lib/oeqa/selftest/meta-refkit-extra.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+# ex:ts=4:sw=4:sts=4:et
+# -*- tab-width: 4; c-basic-offset: 4; indent-tabs-mode: nil -*-
+#
+# Copyright (c) 2017, Intel Corporation.
+# All rights reserved.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# AUTHORS
+# Patrick Ohly <patrick.ohly@intel.com>
+
+"""
+Test cases that verify certain aspects of meta-refkit-extra
+in relation to meta-refkit. They need to be here because
+meta-refkit-extra is not enabled by default and thus
+tests defined there wouldn't be found.
+"""
+
+import re
+
+from oeqa.selftest.base import oeSelfTest
+from oeqa.utils.commands import runCmd, bitbake, get_bb_var
+
+class MetaRefkitExtra(oeSelfTest):
+    """
+    These tests must run in a build configuration that does *not* have
+    meta-refkit-extra.
+    """
+
+    def setUpLocal(self):
+        """
+        This code is executed before each test method.
+        It verifies that meta-refkit-extra is not active.
+        """
+
+        result = runCmd('bitbake-layers show-layers')
+        for layer, path, pri in re.findall(r'^(\S+)\s+(.*?)\s+(\d+)$', result.output, re.MULTILINE):
+            if layer == 'meta-refkit-extra':
+                self.skipTest('meta-refkit-extra already in bblayers.conf')
+            if layer == 'meta-refkit':
+                self.refkit_extra_path = path + '/../meta-refkit-extra'
+        self.extra_conf = "require conf/distro/include/refkit-extra.conf"
+        self.image_dir = get_bb_var('DEPLOY_DIR_IMAGE')
+        self.machine = get_bb_var('MACHINE')
+
+    def enable_refkit_extra(self):
+        '''
+        Adds the meta-refkit-extra layer and enables its content.
+        Will be undone after the test completes.
+        '''
+        self.write_bblayers_config('REFKIT_LAYERS += "%s"' % self.refkit_extra_path)
+        self.write_config(self.extra_conf)
+
+    def test_yocto_compat(self):
+        """
+        Checks for 'Yocto Compatible 2.0' using the yocto-compat-layer.py tool.
```

Comment:
You could open this up in the comment a bit. What is "Yocto Compatible 2.0" and where is it defined? What does it mean if the script succeeds?